app/javascript/pages/country/widget/widgets/widget-fao-forest/widget-fao-forest-selectors.js

```diff
@@ -0,0 +1,78 @@
+import { createSelector } from 'reselect';
+import { format } from 'd3-format';
+
+// get list data
+const getFAO = state => state.fao || null;
+const getRank = state => state.rank || null;
+const getLocationNames = state => state.locationNames || null;
+
+// get lists selected
+export const getFAOForestData = createSelector(
+  [getFAO, getRank, getLocationNames],
+  (fao, rank, locationNames) => {
+    if (!fao || !rank || !locationNames) return null;
+
+    const {
+      area_ha,
+      extent,
+      forest_planted,
+      forest_primary,
+      forest_regenerated
+    } = fao;
+
+    const naturallyRegenerated = extent / 100 * forest_regenerated;
+    const primaryForest = extent / 100 * forest_primary;
+    const plantedForest = extent / 100 * forest_planted;
+    const nonForest =
+      area_ha - (naturallyRegenerated + primaryForest + plantedForest);
+    const total =
+      naturallyRegenerated + primaryForest + plantedForest + area_ha;
+    return {
+      chartData: [
+        {
+          name: 'Naturally regenerated Forest',
+          value: naturallyRegenerated,
+          percentage: naturallyRegenerated / total * 100,
+          color: '#959a00',
+          legendValue: format('.1f')(naturallyRegenerated / total * 100)
+        },
+        {
+          name: 'Primary Forest',
+          value: primaryForest,
+          percentage: primaryForest / total * 100,
+          color: '#2d8700',
+          legendValue: format('.1f')(primaryForest / total * 100)
+        },
+        {
+          name: 'Planted Forest',
+          value: plantedForest,
+          percentage: plantedForest / total * 100,
+          color: '#1e5a00',
+          legendValue: format('.1f')(plantedForest / total * 100)
+        },
+        {
+          name: 'Non-Forest',
+          value: nonForest,
+          percentage: nonForest / total * 100,
+          color: '#d1d1d1',
+          legendValue: format('.1f')(nonForest / total * 100)
+        }
+      ],
+      sentence: {
+        __html: `FAO data from 2015 shows that ${locationNames.current &&
+          locationNames.current.label} is ${
+          nonForest / area_ha > 0.5 ? 'mostly non-forest.' : 'mostly forest.'
+        }${
+          primaryForest > 0
+            ? ` Primary forest occupies <strong>${format('.1f')(
```

Comment:
Again, this is a data selector, not a render so wondering whether having format and sentence builders here is sensible. Thoughts?